feat: manage page support outcome, note, review creation

JohnsonMao opened the pull request:
- 計畫管理頁 - 成果、便利貼、復盤的新增功能
- 重構 service 與 swr 的使用方式
- 調整 Task 樣式

JohnsonMao committed:
ci: update workflow trigger conditions for dev branch

JohnsonMao committed:
feat: implement project selection modal and review creation

- Add SelectProjectModal component for project selection
- Rename mutation methods in useProjectReview for better clarity
- Integrate project selection and review creation in manage page
- Enhance Modal component to support more flexible content rendering

JohnsonMao committed:
refactor(hooks): rename mutation functions for better clarity

- Rename create/update/remove to createMutation/updateMutation/deleteMutation in project hooks
- Add Note and Outcome form components to manage page
- Refactor success handlers for better code reuse
- Adjust SelectProjectModal size from lg to md
- Uncomment onRemovedDOM handler in SelectProjectModal

JohnsonMao committed:
refactor: rename mutation variables for better clarity and consistency

- Rename mutation variables in useProjectMilestone and useProjectTask hooks
  - 'create' -> 'createMutation'
  - 'update' -> 'updateMutation'
  - 'remove' -> 'deleteMutation'
- Remove redundant 'revalidateIfStale' option from SWR config
- Temporarily disable 'new task' feature with error toast

JohnsonMao committed:
refactor(services): move httpClient to core directory

JohnsonMao committed:
refactor(comments): restructure comment service architecture

- Move comment functionality from hooks/api to services/modules
- Create dedicated schema definitions with zod validation
- Improve API organization with better separation of concerns
- Rename hook methods for better semantic clarity (create → createMutation)
- Add user schema definitions to support comment functionality
- Update CommentSection component to use the new service structure

vincentxuu commented:
/gemini review

vincentxuu commented on hooks/api/project/useProjectMilestone.ts:
```diff
@@ -36,14 +36,14 @@ export default function useProjectMilestone({
   const { data, ...swr } = useSWR<ProjectMilestoneSchema>(swrKey);
 
-  const create = useSWRMutation(
+  const createMutation = useSWRMutation(
```
好奇這邊是語意化修正嗎?

JohnsonMao replied:
對～之前那樣命名我覺得不太好ＸＤ

vincentxuu commented on shared/components/Comment/CommentSection.tsx:
```diff
@@ -12,10 +11,10 @@ interface CommentSectionProps {
-    update: updateComment,
-    remove: removeComment,
-  } = useCommentList({
+    createMutation,
```
我在寫idea使用也有這個困擾，型態與function為什麼要分開命名，不能一起使用，沒想到其實可以..
我看到create就覺得是一個function，就直覺得直接拿來用....

const { data: idea, update: onUpdated, remove: onDeleted } = useIdea({

JohnsonMao replied:
我有點看不懂你在說什麼＠＠

vincentxuu replied:
update: onUpdated

會覺得我用update就應該可以代表這個function

vincentxuu commented on shared/components/Modal.tsx:
```diff
@@ -12,7 +12,7 @@ enum ModalSize {
 export interface ModalProps {
   isOpen: boolean;
-  title?: string;
+  title?: React.ReactNode;
```
string>>React.ReactNode好奇有什麼特殊用途嗎?

JohnsonMao replied:
讓 title 也可以直接傳 React 元件～而不侷限於字串

vincentxuu replied:
學習了，那具體來說還有什麼可能可以被傳入?

noracami replied:
例如一個自帶 style 的字串 🤣

vincentxuu replied:
這樣好嗎><

JohnsonMao commented:
先不要合唷～我還在重構～

JohnsonMao committed:
feat(services): add PathBuilder for API path management

Add a new PathBuilder class that provides a fluent interface for constructing
API paths with proper encoding.

JohnsonMao committed:
refactor(comments): restructure API and hooks architecture

- Refactor comment API to use typed mutation fetchers
- Convert hooks to standardized SWR key pattern
- Simplify endpoint construction with apiPaths utility
- Update components to use new API structure

JohnsonMao committed:
refactor(projects): migrate project services to modular structure

- Move project-related APIs and hooks from hooks/api/project to services/modules/projects
- Reorganize project services into core, milestones, and tasks modules
- Standardize naming conventions (endpoint → pathname, Request → Schema)
- Update components to use the new service structure
- Improve type safety with proper null handling for projectId
- Enhance code organization with better separation of concerns

JohnsonMao committed:
refactor(notes): migrate project notes API to modular structure

- Move notes API from services/projects to services/modules/projects/notes
- Rename type definitions from *Request to *Schema for consistency
- Rename getProjectNoteEndpoint to getProjectNotePathname
- Improve parameter validation with zod
- Update all imports and usages across components and pages

JohnsonMao committed:
refactor: enhance Note with multi-image support

- Update Upload component API from onPreviewChange to onPreviewsChange
- Refactor Note Form to support multiple image uploads with preview and deletion
- Consolidate project service imports from separate modules to centralized path
- Standardize type naming convention using Schema suffix consistently

JohnsonMao committed:
feat: add image upload functionality

- Create images module with API and schema
- Refactor httpClient serialization logic
- Update Upload component to handle image validation

JohnsonMao committed:
feat: enhance multi-image upload and display functionality

- Refactor outcome API services for better organization
- Implement multiple image upload with preview and delete
- Standardize parameter parsing across detail pages

JohnsonMao committed:
refactor: migrate project review APIs to services module

- Move review APIs from hooks/api to services/modules/projects
- Standardize API structure with other project modules
- Improve code readability with consistent patterns

JohnsonMao committed:
refactor(milestone): optimize MilestoneCard component

- Improve form handling with proper values and defaultValues
- Replace TaskList with direct Task component rendering
- Add date validation to milestone schemas

JohnsonMao commented:
/gemini review

JohnsonMao committed:
refactor: consolidate project API services into modules

- Remove individual API service files in favor of modules approach
- Update import paths across components and pages
- Change dev port from 443 to 5438
